AWSWindowsHelpers/public/ARN/Get-AWSWindowsHelpersARNDetail.ps1

```diff
@@ -0,0 +1,128 @@
+<#
+.Synopsis
+   Returns the details of an AWS Resource from the ARN
+.DESCRIPTION
+   Returns the details of an AWS Resource from the ARN
+.EXAMPLE
+   Get-AWSWindowsHelpersARNDetail -arn "arn:aws:iam::123456789012:server-certificate/2017_wild_example_com"
+.INPUTS
+   arn - ARN of the AWS certificate
+.OUTPUTS
+   Returns a object with the following properties
+   Partition - The partition the resource is in
+   Service - The AWS Service
+   Region - The AWS region
+   AccountID - AWS account number
+   ResourceDetail - The resource ARN detail
+   ResourceType - The type of AWS resource
+.FUNCTIONALITY
+   Retrieves from a ARN the details of the AWS service
+#>
+Function Get-AWSWindowsHelpersARNDetail
+{
+    [CmdletBinding()]
+    Param(
+        [Parameter(Mandatory=$true)]
+        [ValidatePattern('^arn:*.:*.:*.:*.:*.')]
+        $arn      
+    )
+    $arnSplit = $arn -split ":"
+
+    $arnDetail = @{
+        partition = $arnSplit[1]
+        service = $arnSplit[2]
+        region = $arnSplit[3]
+        accountid = $arnSplit[4]
+    }
+
+    $resourceDetail = $arnSplit[5..$arnSplit.Length] -join ":"
+    $arnDetail.Add("resourcedetail",$resourceDetail)
+    $resourceARN = $resourceDetail -split "/"
+    $resourceList= @()
+
+    if($arnSplit.Length -le 6 -and $($resourceARN.length -eq 1) -or $arnDetail.service -eq 's3')
+    {   # Deal with ARN in format "arn:partition:service:region:account-id:resource"
+        if($($resourceARN.length -gt 1))
+        { # Handle S3 resources
+            $arnDetail.Add("resource",$resourceARN[0])
+        }
+        else 
+        {
+            $arnDetail.Add("resource",$arnSplit[5])    
+        }
+        Write-verbose "Adding [resource] $($arnDetail.resource)"
+    }
+    elseif($arnSplit.Length -ge 6 -and $($resourceARN.length -ge 1))
+    {   # Deal with ARN in format "arn:partition:service:region:account-id:resourcetype/resource"
+        if($resourceARN[0] -match ":")
+        {
+            $resourceType = $($resourceARN[0] -split ":")[0]
+        }
+        else 
+        {
+            $resourceType = $resourceARN[0]
+        }
+        $arnDetail.Add("resourcetype",$resourceType)
+        for($element=5;$element -lt $arnSplit.Length;$element++)
```

Comment:
Would something like
`foreach($element in $arnsplit[5..$arnsplit.length])`
be easier to read?

Reply:
I'm using the `$element` value to to look at adjacent element in the array to determine if this is a key value pair separated by a `:` (e.g `arn:aws:cloud9:region:account-id:environment:environment-id` or `arn:aws:logs:us-east-1:123456789012:log-group:my-log-group:log-stream:my-log-stream` in the ARN instead of a `/`